llvm-21.yaml

```diff
@@ -0,0 +1,917 @@
+package:
+  name: llvm-21
+  version: "21.1.1"
+  epoch: 0
+  description: Low-level virtual machine ${{vars.major-version}} - core frameworks
+  copyright:
+    - license: Apache-2.0
+  resources:
+    cpu: 32
+    memory: 64Gi
+  dependencies:
+    provides:
+      - llvm=${{package.full-version}}
+
+vars:
+  llvm-prefix: "/usr/lib/llvm-21"
+  clang-prefix: "/usr/lib/llvm-21/lib/clang/21"
```

Comment:
I know they are like this in -20 but both of these could be generated in var-transforms, avoiding a hardcode of 21 - will help when we do 22 :)